astropy/utils/data.py

```diff
@@ -1391,6 +1391,36 @@ def _deltemps():
                     pass
 
 
+def _remove_download_cache(pkgname='astropy'):
+    zapped_cache = False
+    try:
+        with contextlib.ExitStack() as stack:
+            try:
+                dldir, url2hash = stack.enter_context(
+                    _cache(pkgname, write=True))
+            except (RuntimeError,     # Couldn't get lock
+                    WrongDBMModule):  # Incompatible cache
+                # Release lock by blowing away cache
+                # Need to get locations
+                dldir, _ = _get_download_cache_locs(pkgname)
+                url2hash = None
+            if os.path.exists(dldir):
+                # This can be awkward if the shelve is still open
+                # NFS can't delete an open file
```

Comment:
This comment isn't completely accurate. NFS _can_ delete an open file, but cannot delete a directory that contains the file while it is still open.